02_configure_host.sh

```diff
@@ -164,8 +164,49 @@ if [ "$MANAGE_INT_BRIDGE" == "y" ] || [ "$MANAGE_PRO_BRIDGE" == "y" ]; then
   sudo systemctl enable network
 fi
 
+# VLAN interface in the provisioning bridge
+if [ ! -z "${PROVISIONING_VLAN}" ] ; then
+    VLAN_INT=${PROVISIONING_NETWORK_NAME}.${PROVISIONING_VLAN}
+    sudo cp /var/lib/libvirt/dnsmasq/${BAREMETAL_NETWORK_NAME}.conf /etc/NetworkManager/dnsmasq.d/${VLAN_INT}.conf
+    sudo sed -i "s/${BAREMETAL_NETWORK_NAME}/${VLAN_INT}/" /etc/NetworkManager/dnsmasq.d/${VLAN_INT}.conf
+    sudo sed -i "/^bind-dynamic/d" /etc/NetworkManager/dnsmasq.d/${VLAN_INT}.conf
+    sudo sed -i 's#/var/lib/libvirt/dnsmasq/#/etc/NetworkManager/dnsmasq-shared.d/#' /etc/NetworkManager/dnsmasq.d/${VLAN_INT}.conf
+    sudo cp /var/lib/libvirt/dnsmasq/${BAREMETAL_NETWORK_NAME}.hostsfile /etc/NetworkManager/dnsmasq-shared.d/${VLAN_INT}.hostsfile
+    sudo cp /var/lib/libvirt/dnsmasq/${BAREMETAL_NETWORK_NAME}.addnhosts /etc/NetworkManager/dnsmasq-shared.d/${VLAN_INT}.addnhosts
+
+    # Remove the baremetal libvirt network
+    sudo virsh net-destroy ${BAREMETAL_NETWORK_NAME}
+    sudo virsh net-undefine ${BAREMETAL_NETWORK_NAME}
+
+    # Remove the baremetal bridge
+    sudo ifdown ${BAREMETAL_NETWORK_NAME} || true
+    if [ -e /etc/sysconfig/network-scripts/ifcfg-${BAREMETAL_NETWORK_NAME} ] ; then
+        sudo rm -f /etc/sysconfig/network-scripts/ifcfg-${BAREMETAL_NETWORK_NAME}
+    fi
+
+    # Remove the baremetal bridge definitions in the VMs
+    DOMLIST=$(sudo virsh list --all | grep ${CLUSTER_NAME} | awk '{print $2}')
+    for VM in ${DOMLIST}; do
+        EXT_BR_MAC=$(sudo virsh domiflist ${VM} | grep ${BAREMETAL_NETWORK_NAME} | awk '{print $5}')
+        sudo virsh detach-interface $VM bridge --mac ${EXT_BR_MAC} --config
```

Comment:
I don't think we want to detach the interface, we need to edit the VM definition so that the previous baremetal nic is connected to the provisioning bridge?

E.g something like `sudo virt-xml --edit all ostest_worker_1 --network bridge=ostestpr` should work?

@mcornea I notice in the QE automation you did `virt-xml --remove-device` then add the devices back with an updated configuration, was there a specific reason for that approach?

Reply:
No specific reason, --remove/add-device were the options I was aware of at the time to update the NIC devices attached to the VMs.